From cc8850d74d6a7441867751c5807ca20f677f8992 Mon Sep 17 00:00:00 2001
From: Brad Barnett <bbarnett@gmail.com>
Date: Mon, 27 Jul 2026 23:44:22 -0500
Subject: [PATCH 2/2] cmods-mp-patch: raise unix MICROPY_SCHEDULER_DEPTH for
 desktop SDL.

usdl2/pydisplay timers enqueue via mp_sched_schedule; depth 4 overflows.
Must be a power of 2. Applied by cmods build_mp.sh.
---
 ports/unix/mpconfigport.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ports/unix/mpconfigport.h b/ports/unix/mpconfigport.h
index 0f4ee9765..e790fca0c 100644
--- a/ports/unix/mpconfigport.h
+++ b/ports/unix/mpconfigport.h
@@ -111,6 +111,12 @@ typedef long mp_off_t;
 #define MICROPY_GCREGS_SETJMP (1)
 #endif
 
+// Desktop SDL timers (usdl2 / pydisplay) enqueue via mp_sched_schedule; depth 4
+// overflows under auto-refresh + app timers. Must be a power of 2.
+#ifndef MICROPY_SCHEDULER_DEPTH
+#define MICROPY_SCHEDULER_DEPTH     (16)
+#endif
+
 // Enable the VFS, and enable the posix "filesystem".
 #define MICROPY_ENABLE_FINALISER    (1)
 #define MICROPY_VFS                 (1)
-- 
2.43.0

